app/lib/actions/pulp3/repository/update_remote.rb

```diff
@@ -0,0 +1,16 @@
+module Actions
+  module Pulp3
+    module Repository
+      class UpdateRemote < Pulp3::Abstract
+        def plan(repository, smart_proxy)
+          plan_self(:repository_id => repository.id, :smart_proxy_id => smart_proxy.id) if repository.root.url?
```

Comment:
The update_remote method below will delete the remote if its no longer needed, but that doesn't happen because of the if check